use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,484 @@
+reinstalled. By doing it this way you no longer have to rely on your 
+distribution to include the Intel QAT firmware. The modules will load every
+reboot and the qat_service helper script gets added. 
+
+```sh
+$ cd $QAT_SRC
+$ ./configure --disable-qat-lkcf  
+$ sudo -E make install -j$(nproc)
+$ sudo chown -R $USER:$USER $QAT_SRC
+```
+
+### Copy Intel QAT configuration files and enable Virtual Functions
+
+Learn more about customizing configuration files at the 
+[Intel QAT Engine github](https://github.com/intel/QAT_Engine/#copy-the-correct-intel-quickassist-technology-driver-config-files)
+This section starts from a base config file and changes the `SSL` section to 
+`SHIM` to support the OpenSSL engine. There are many more tweaks that could be 
+made depending on the use case and how many Intel QAT engines should be run. 
+More information about how to customize is located in the 
+[Intel® QuickAssist Technology Software for Linux* - Programmer's Guide.](https://01.org/sites/default/files/downloads/336210qatswprogrammersguiderev006.pdf) 
+
+```sh
+$ mkdir -p $QAT_CONF_LOCATION
+$ cp $QAT_SRC/build/c6xxvf_dev0.conf.vm $QAT_CONF_LOCATION/c6xxvf_dev0.conf
+$ sed -i 's/\[SSL\]/\[SHIM\]/g' $QAT_CONF_LOCATION/c6xxvf_dev0.conf
+$ for i in {1..15}; do cp $QAT_CONF_LOCATION/c6xxvf_dev0.conf $QAT_CONF_LOCATION/c6xxvf_dev$i.conf; done
+```
+
+### Expose and Bind Intel QAT virtual functions to VFIO-PCI (Every reboot)
+
+To enable virtual functions, the host OS should have IOMMU groups enabled. In 
+the UEFI Firmware Intel Virtualization Technology for Directed I/O 
+(Intel VT-d) must be enabled. Also, the kernel boot parameter should be 
+`intel_iommu=on` or `intel_iommu=ifgx_off`. The default in Clear Linux currently 
+is `intel_iommu=igfx_off` which should work with the Intel QAT device. The 
+following commands assume an Intel QAT card is installed, IOMMU is on, and 
+VT-d is enabled. The vendor and device ID will be added to the `VFIO-PCI` driver
+so that each exposed virtual function can be bound to the `VFIO-PCI` driver. 
+Once that is done then each virtual function can be passed into a Kata 
+Containers container using the PCIe device passthrough feature. For 
+Kubernetes, the Intel device plugin for Kubernetes handles the binding of 
+the driver but the VF’s still must be enabled.
+
+```sh
+$ sudo modprobe vfio-pci
+$ QAT_PCI_BUS_PF_NUMBERS=$(lspci | grep Quick | cut -d ' ' -f 1)
+$ QAT_PCI_BUS_PF_1=$(echo $QAT_PCI_BUS_PF_NUMBERS | cut -d ' ' -f 1)
+$ echo 16 | sudo tee /sys/bus/pci/devices/0000:$QAT_PCI_BUS_PF_1/sriov_numvfs
+$ QAT_PCI_ID_VF=$(cat /sys/bus/pci/devices/0000:${QAT_PCI_BUS_PF_1}/virtfn0/uevent | grep PCI_ID)
+$ QAT_VENDOR_AND_ID_VF=$(echo ${QAT_PCI_ID_VF/PCI_ID=} | sed 's/:/ /')
+$ echo $QAT_VENDOR_AND_ID_VF | sudo tee --append /sys/bus/pci/drivers/vfio-pci/new_id
+
+# Loop through all the virtual functions and bind to the VFIO driver
+$ for f in /sys/bus/pci/devices/0000:$QAT_PCI_BUS_PF_1/virtfn*
```

Comment:
Missing opening ` ```sh ` for this block.

Reply:
I broke this up into more modular code blocks.